lib/Document.js

```diff
@@ -294,10 +294,11 @@ Document.prototype = Object.create(Node.prototype, {
   adoptNode: { value: function adoptNode(node) {
     if (node.nodeType === Node.DOCUMENT_NODE) utils.NotSupportedError();
 
-    if (node.parentNode) node.parentNode.removeChild(node);
```

Comment:
Please always check specification 
https://www.w3.org/TR/2004/REC-DOM-Level-3-Core-20040407/core.html#Document3-adoptNode

> Note that if the adopted node is already part of this document (i.e. the source and target document are the same), this method still has the effect of removing the source node from the child list of its parent,

and run the tests !!.